test/datasets.js

```diff
@@ -223,6 +227,84 @@ describe('Dataset', () => {
                 .then(given => expect(given).to.be.eql(expected));
         });
 
+        describe('getData()', () => {
+            const message = 'CUSTOM ERROR';
+            it('getData() should return null for 404', async () => {
+                const requestPromise = () => {
+                    throw new ApifyClientError('NOTFOUND', 'Not found', { statusCode: NOT_FOUND_STATUS_CODE });
+                };
+
+                const data = await getData(requestPromise);
+                expect(data).to.eql(null);
+            });
+
+            it('parseResponse() should rethrow errors', async () => {
+                const response = {
+                    headers: {
+                        'content-type': 'application/json',
+                    },
+                    body: {
+
+                    },
+                };
+                const stub = sinon.stub(utils, 'parseBody');
+                stub.callsFake(() => {
+                    throw new Error(message);
+                });
+                let error;
+                try {
+                    parseResponse(response, false);
+                } catch (e) {
+                    error = e;
+                }
+                expect(error instanceof Error).to.be.eql(true);
+                expect(error.message).to.be.eql(message);
+                utils.parseBody.restore();
+            });
+
+            it('parseResponse() should throw RetryableError if  Unexpected end of JSON input', async () => {
+                const response = {
+                    headers: {
+                        'content-type': 'application/json',
+                    },
+                    body: {
+
+                    },
+                };
+                const stub = sinon.stub(utils, 'parseBody');
+                stub.callsFake(() => {
+                    throw new Error('Unexpected end of JSON input');
```

Comment:
Hey what about to create that error this way 👍 to ensure that has correct format

```
stub.callsFake(() => {
    JSON.parse('{"foo":')
});
```